crates/web5/src/crypto/dsa/ed25519.rs

```diff
@@ -109,29 +117,347 @@ impl Ed25519Verifier {
 }
 
 impl Verifier for Ed25519Verifier {
-    fn verify(&self, payload: &[u8], signature: &[u8]) -> Result<bool> {
+    fn verify(&self, payload: &[u8], signature: &[u8]) -> Result<()> {
+        if let Some(d) = &self.public_jwk.d {
+            if !d.is_empty() {
+                return Err(Web5Error::Crypto(
+                    "provided verification key cannot contain private key material".to_string(),
+                ));
+            }
+        }
+
         let mut public_key_bytes = [0u8; PUBLIC_KEY_LENGTH];
         let decoded_x = general_purpose::URL_SAFE_NO_PAD.decode(&self.public_jwk.x)?;
 
         if decoded_x.len() != PUBLIC_KEY_LENGTH {
-            return Err(DsaError::InvalidKeyLength(PUBLIC_KEY_LENGTH.to_string()));
+            return Err(Web5Error::Crypto(format!(
+                "invalid public key length {} must be {}",
+                decoded_x.len(),
+                PUBLIC_KEY_LENGTH
+            )));
         }
 
         public_key_bytes.copy_from_slice(&decoded_x);
         let verifying_key = VerifyingKey::from_bytes(&public_key_bytes)
-            .map_err(|e| DsaError::PublicKeyFailure(e.to_string()))?;
+            .map_err(|e| Web5Error::Crypto(format!("unable to instantiate verifying key {}", e)))?;
 
         if signature.len() != SIGNATURE_LENGTH {
-            return Err(DsaError::InvalidSignatureLength(self.public_jwk.x.clone()));
+            return Err(Web5Error::Crypto(format!(
+                "invalid signature length {} must be {}",
+                signature.len(),
+                SIGNATURE_LENGTH
+            )));
         }
 
         let mut signature_bytes = [0u8; SIGNATURE_LENGTH];
         signature_bytes.copy_from_slice(signature);
         let verify_result = verifying_key.verify(payload, &Signature::from_bytes(&signature_bytes));
 
         match verify_result {
-            Ok(_) => Ok(true),
-            Err(_) => Ok(false),
+            Ok(_) => Ok(()),
+            Err(_) => Err(Web5Error::Crypto(
+                "cryptographic verification failure".to_string(),
+            )),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{test_helpers::UnitTestSuite, test_name};
+    use general_purpose::URL_SAFE_NO_PAD;
+    use std::sync::LazyLock;
+
+    mod generate {
+        use super::*;
+
+        static TEST_SUITE: LazyLock<UnitTestSuite> =
+            LazyLock::new(|| UnitTestSuite::new("ed25519_generate"));
```

Comment:
Are we committing to using rustc 1.80.0 features in our tests for this crate?

Reply:
no def not, I'm just behind on your MSRV work